pkg/noderesourcetopology/filter_test.go

```diff
@@ -339,6 +339,233 @@ func TestNodeResourceTopology(t *testing.T) {
 	}
 }
 
+type resourceDescriptor struct {
+	Host     string
+	Node     string
+	Resource string
+	Quantity string
+}
+
+func TestNodeResourceTopologyMultiContainerPodScope(t *testing.T) {
+	nodeTopologies := []*topologyv1alpha1.NodeResourceTopology{
+		{
+			ObjectMeta:       metav1.ObjectMeta{Name: "host0"},
+			TopologyPolicies: []string{string(topologyv1alpha1.SingleNUMANodePodLevel)},
+			Zones: topologyv1alpha1.ZoneList{
+				{
+					Name: "node-0",
+					Type: "Node",
+					Resources: topologyv1alpha1.ResourceInfoList{
+						MakeTopologyResInfo(cpu, "32", "30"),
+						MakeTopologyResInfo(memory, "64Gi", "60Gi"),
+						MakeTopologyResInfo(hugepages2Mi, "384Mi", "384Mi"),
+						MakeTopologyResInfo(nicResourceName, "16", "16"),
+					},
+				},
+				{
+					Name: "node-1",
+					Type: "Node",
+					Resources: topologyv1alpha1.ResourceInfoList{
+						MakeTopologyResInfo(cpu, "32", "32"),
+						MakeTopologyResInfo(memory, "64Gi", "64Gi"),
+						MakeTopologyResInfo(hugepages2Mi, "512Mi", "512Mi"),
+						MakeTopologyResInfo(nicResourceName, "32", "32"),
+					},
+				},
+			},
+		},
+	}
+
+	nodes := make([]*v1.Node, len(nodeTopologies))
+	for i := range nodes {
+		nodes[i] = makeNodeFromNodeResourceTopology(nodeTopologies[i])
+	}
+
+	tests := []struct {
+		name       string
+		pod        *v1.Pod
+		node       *v1.Node
+		nrts       []*topologyv1alpha1.NodeResourceTopology
+		avail      []resourceDescriptor
+		wantStatus *framework.Status
+	}{
+		{
+			name: "gu pod fits only on a numa node",
+			pod: makeMultiContainerPodWithResourceList("testpod",
+				[]v1.ResourceList{
+					{
+						v1.ResourceCPU:    resource.MustParse("2"),
+						v1.ResourceMemory: resource.MustParse("2Gi"),
+					},
+					{
+						v1.ResourceCPU:    resource.MustParse("4"),
+						v1.ResourceMemory: resource.MustParse("8Gi"),
+					},
+					{
+						v1.ResourceCPU:                   resource.MustParse("26"),
+						v1.ResourceMemory:                resource.MustParse("32Gi"),
+						v1.ResourceName(hugepages2Mi):    resource.MustParse("512Mi"),
+						v1.ResourceName(nicResourceName): resource.MustParse("26"),
+					},
+				},
+			),
+			node: nodes[0],
+			nrts: []*topologyv1alpha1.NodeResourceTopology{
+				nodeTopologies[0],
+			},
+			avail:      []resourceDescriptor{},
+			wantStatus: nil,
+		},
+		{
+			name: "gu pod does not fit - CPUs",
```

Comment:
A borderline NIT but would be nice to make the test case name more descriptive:
```suggestion
			name: "gu pod does not fit - not enough CPUs available on any NUMA node",
```

Reply:
done